### PR TITLE
Add EdgeInsets, MediaQuery support for view insets

### PR DESCRIPTION
**Pull request by cbracken**
Adds EdgeInsets.fromViewInsets constructor, and
MediaQueryData.viewInsets.

See engine commit flutter/engine#4403

### CODE REVIEW

**Comment by Hixie** [comment]
Why not just reuse WindowPadding?

File: packages/flutter/lib/src/widgets/media_query.dart [hunk 1]
```diff
@@ -76,6 +78,9 @@ class MediaQueryData {
   /// the specified font size.
   final double textScaleFactor;
 
+  /// The system insets around the edges of the media (e.g., the screen).
+  final EdgeInsets viewInsets;
```
Comment (xster) [inline packages/flutter/lib/src/widgets/media_query.dart]: this is the most user facing front document right? Maybe add a bit more details for the nuances between this and padding

**Review by xster — APPROVED** [review]
LG

**Comment by cbracken** [comment]
> Why not just reuse WindowPadding?

@Hixie Done in flutter/engine#4406 (with minor doc updates for that class)

File: packages/flutter/lib/src/widgets/media_query.dart (hunk 1 above)
Comment (cbracken) [reply packages/flutter/lib/src/widgets/media_query.dart]: Good point! Done.

**Comment by xster** [comment]
LG

**Pull request merged**